### PR TITLE
Add basic hashing functionality for enclave Hasher

### PR DESCRIPTION
**Pull request by lkatalin**
Co-authored by: Nathaniel McCallum <npmccallum@redhat.com>

### CODE REVIEW

File: sgx-hasher/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+version = "0.1.0"
+authors = ["Lily Sturmann <lsturman@redhat.com>", "Nathaniel McCallum <npmccallum@redhat.com>" ]
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+sgx-types = { path = "../sgx-types" }
+sgx-traits = { path = "../sgx-traits" }
+sgx-sys = { path = "../sgx-sys" }
```
Comment (npmccallum) [inline sgx-hasher/Cargo.toml]: This dependency does not appear to be needed.

File: sgx-hasher/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        &mut self,
+        offset: usize,
+        data: &[u8],
+        flags: sgx_traits::Flags,
+        secinfo: page::SecInfo,
+    ) {
+        const EEXTEND: u64 = 0x00444E4554584545;
+        const EADD: u64 = 0x0000000044444145;
+
+        let si = secinfo.as_ref();
```
Comment (npmccallum) [inline sgx-hasher/src/lib.rs]: I don't think we need a separate variable for this. Let's do `&secinfo.as_ref()[..48]` below.

File: sgx-hasher/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                self.0.update(&EEXTEND.to_le_bytes());
+                self.0.update(&eoff.to_le_bytes());
+                self.0.update(&[0u8; 48]);
+
+                self.0.update(&data[poff..][..256]);
+            }
+        }
+    }
+
+    // Returns SHA256 hash of MRENCLAVE
```
Comment (npmccallum) [inline sgx-hasher/src/lib.rs]: This text is confusing. It seems to imply that `MRENCLAVE` is being hashed. Rather, this function is producing `MRENCLAVE`.

File: sgx-traits/src/lib.rs [hunk 4]
```diff
@@ -16,7 +16,12 @@ use std::io::Result;
 
 use bitflags::bitflags;
 use sgx_types::page::{Flags as PageFlags, SecInfo};
-use sgx_types::{secs::Secs, sig::Signature, tcs::Tcs, Offset};
+use sgx_types::{
+    secs::Secs,
+    sig::Signature,
+    tcs::Tcs,
+    Offset,
+};
```
Comment (npmccallum) [inline sgx-traits/src/lib.rs]: What is the purpose of this change?

File: sgx-hasher/src/lib.rs (hunk 3 above)
Comment (lkatalin) [reply sgx-hasher/src/lib.rs]: That's a good point.

File: sgx-traits/src/lib.rs (hunk 4 above)
Comment (lkatalin) [reply sgx-traits/src/lib.rs]: It's `cargo fmt`. In general I'm still not sure if `cargo fmt` changes should be their own PR... that seems sort of cluttered for the `git log`? But otherwise it ends up getting folded in, like this. Open to suggestions.

Comment (npmccallum) [reply sgx-traits/src/lib.rs]: The question is why is `cargo fmt` making this change? The previous commits had `cargo fmt` run on them. So we shouldn't be seeing this.

Comment (npmccallum) [reply sgx-traits/src/lib.rs]: I'll merge this PR and figure it out afterwards.

Comment (lkatalin) [reply sgx-traits/src/lib.rs]: I just ran it again and it went away, so I'm confused about it too. Let me fix it and I'll re-push.

**Comment by npmccallum** [comment]
@lkatalin Tests are failing. Also, lots of warnings are being introduced. Please don't add new warnings.

**Comment by lkatalin** [comment]
@npmccallum Odd, I'll check on these and fix them.
